src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -93,6 +93,34 @@ internal static class ConstraintsHelper
             }
 
             diagnosticsBuilder.Free();
+
+            if (typeParameter.AllowByRefLike)
+            {
+                if (inherited)
+                {
+                    Location location = typeParameter.GetFirstLocation();
```

Comment:
I got the impression that getting location unnecessarily might be expensive - if that's true, could we pass typeParameter to the diagnostics instead?

Reply:
> I got the impression that getting location unnecessarily might be expensive - if that's true, could we pass typeParameter to the diagnostics instead?

There is no overload that takes TypeParameterSymbol. A couple of conditions should be met before we get here. I don't think the situation is going to be very common and I prefer to not complicate the code until we have data indicating that there is a problem to fix.